state/machine.go

```diff
@@ -2203,6 +2182,78 @@ func (m *Machine) RemoveUpgradeSeriesLock() error {
 	return nil
 }
 
+// UpgradeSeriesStatus returns the status of a series upgrade.
+func (m *Machine) UpgradeSeriesStatus(unitName string) (params.UnitSeriesUpgradeStatus, error) {
+	coll, closer := m.st.db().GetCollection(machineUpgradeSeriesLocksC)
```

Comment:
can lines 2187-2198 go in a helper function to return the lock data?  They are used by IsLocked() and will be used for future work as well.

Reply:
I would want to wait a bit to pull this out into a helper, perhaps we can wait until more of a tell-don't-ask pattern appears rather than making a helper that facilitates an "ask for lock, then do something with it" pattern.